src/components/item/publish/PublicationStatusComponent.tsx

```diff
@@ -0,0 +1,93 @@
+import CloudDoneIcon from '@mui/icons-material/CloudDone';
+import CloudOffIcon from '@mui/icons-material/CloudOff';
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import ErrorIcon from '@mui/icons-material/Error';
+import EventBusyIcon from '@mui/icons-material/EventBusy';
+import PendingActionsIcon from '@mui/icons-material/PendingActions';
+import PublicOffIcon from '@mui/icons-material/PublicOff';
+import { Chip, CircularProgress } from '@mui/material';
+
+import { PackedItem } from '@graasp/sdk';
+
+import { useBuilderTranslation } from '@/config/i18n';
+import { buildPublicationStatus } from '@/config/selectors';
+import { BUILDER } from '@/langs/constants';
+import { PublicationStatus } from '@/types/publication';
+
+import usePublicationStatus from '../../hooks/usePublicationStatus';
+
+type Props = {
+  item: PackedItem;
+};
+
+export const PublicationStatusComponent = ({ item }: Props): JSX.Element => {
+  const { t } = useBuilderTranslation();
+  const { status, isinitialLoading } = usePublicationStatus({ item });
+
+  if (isinitialLoading) {
+    return (
+      <Chip
+        icon={<CircularProgress size={15} />}
+        label={t(BUILDER.LIBRARY_SETTINGS_PUBLICATION_STATUS_LOADING)}
+        variant="outlined"
+        color="info"
+      />
+    );
+  }
+
+  const chipMap = {
+    [PublicationStatus.Published]: {
```

Comment:
Would it make sense to have some time to ensure all status are defined? I think you used often this structure.